teps/0095-common-repository-configuration.md

```diff
@@ -0,0 +1,251 @@
+- [Proposal](#proposal)
+  - [Notes/Caveats (optional)](#notescaveats-optional)
+  - [Risks and Mitigations](#risks-and-mitigations)
+  - [User Experience (optional)](#user-experience-optional)
+  - [Performance (optional)](#performance-optional)
+- [Design Details](#design-details)
+- [Test Plan](#test-plan)
+- [Design Evaluation](#design-evaluation)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (optional)](#infrastructure-needed-optional)
+- [Upgrade &amp; Migration Strategy (optional)](#upgrade--migration-strategy-optional)
+- [Implementation Pull request(s)](#implementation-pull-request-s)
+- [References (optional)](#references-optional)
+<!-- /toc -->
+
+## Summary
+
+Several projects and designs currently in-flight in the Tekton ecosystem
+are related to or directly working with source repository-related information.
+These include:
+
+- [Scheduled and Polling Runs in Tekton](0083-scheduled-and-polling-runs-in-tekton.md)
+- [Workflows](../working-groups.md#workflows)
+- [Pipeline-as-Code](https://github.com/openshift-pipelines/pipelines-as-code)
+- [Remote Resource Resolution](0060-remote-resource-resolution.md)
+
+Each of these projects are dependent on information from
+repositories - either querying for metadata, fetching content, listening for
+events, recording relatational data about them or publishing notifications to
+them.
+
+Given a shared interest in this data across multiple projects it might
+make sense to define an object that allows an operator to specify common
+repo configuration only once, rather than per-project. Over time other projects
+and tools could also leverage this information.
+
+## Motivation
+
+### Goals
+
+- Decide if there is a common subset of repository data that would be
+  useful across multiple projects.
+- Design a format to store and record that common data.
+- Publish developer-facing docs for projects to use.
+- Publish user-facing docs for operators/cluster admins/app teams to use.
+
+### Non-Goals
+
+- Mutable status. This TEP is not aiming to offer a way for concurrent
+  processess to modify shared state or communicate about state changes.
+  Components can read and respond to the shared Repository configuration
+  data but shouldn't expect access to write back to it.
+- TODO: as part of this TEP review let's figure out the scope and what
+  we'd like to keep as non-goals.
+
+### Use Cases
+
+These use-cases were primarily taken from [the Workflows WG meeting
+notes](https://docs.google.com/document/d/1di4ikeVb8Mksgbq4CzW4m4xUQPZ2dQMLvK1VIJw7OQg/edit#heading=h.i22qgrdiutdu)
+on Nov 16, 2021.
+
+- A component could read info from the common config to decide which branches
+  or commits to poll for updates on, and which API endpoints to hit.
+- A component could read the type, provider and API endpoint info from
+  the common repo config to determine how to convert `CloudEvents` from
+  Pipelines into status updates on that repo.
```

Comment:
Is this something that would need to be done per repo, or per provider?

Reply:
Cheers! Captured the option to have provider / host configuration separate from per-repo information in the Alternatives section.